schematics/utils/ast.ts

```diff
@@ -0,0 +1,58 @@
+import {SchematicsException} from '@angular-devkit/schematics';
+import {Tree} from '@angular-devkit/schematics';
+import * as ts from 'typescript';
+import {addImportToModule} from './devkit-utils/ast-utils';
+import {getAppModulePath} from './devkit-utils/ng-ast-utils';
+import {InsertChange} from './devkit-utils/change';
+import {getConfig, getAppFromConfig} from './devkit-utils/config';
+import {normalize} from '@angular-devkit/core';
+
+/**
+ * Reads file given path and returns TypeScript source file.
+ */
+export function getSourceFile(host: Tree, path: string): ts.SourceFile {
+  const buffer = host.read(path);
+  if (!buffer) {
+    throw new SchematicsException(`Could not find file for path: ${path}`);
+  }
+  const content = buffer.toString();
+  const source = ts.createSourceFile(path, content, ts.ScriptTarget.Latest, true);
+  return source;
+}
+
+/**
+ * Import and add module to root app module.
+ */
+export function addToRootModule(host: Tree, moduleName: string, src: string) {
+  const config = getConfig(host);
+  const app = getAppFromConfig(config, '0');
+  const modulePath = getAppModulePath(host, app);
+  addModuleImportToModule(host, modulePath, moduleName, src);
+}
+
+/**
+ * Import and add module to specific module path.
```

Comment:
Add `@param` JsDoc? This will help disambiguate which module `mouldePath` and `moduleName` are referring to